src/components/map/MapView.svelte

```diff
@@ -78,7 +89,10 @@
 
 		clearAllMarkers();
 
-		if (showRoute && selectedRoute) {
+		if (selectedRoute && !showRoute) {
+			allStops = [];
+			// Add stops to show
```

Comment:
It can be deleted now, but I was thinking that instead of deleting all the stops, we should keep the state of the stops before the search so that when the user finishes the search, the stops reappear. However, I was also considering that we will have a cache, so this shouldn't be a problem.